Install env.sh et al. in etc/ros instead of root prefix

--- cmake/catkin_generate_environment.cmake~	2018-06-06 18:18:22.000000000 +0200
+++ cmake/catkin_generate_environment.cmake	2018-09-04 15:07:25.726401000 +0200
@@ -72,7 +72,7 @@
     @ONLY)
 
   # installspace
-  set(SETUP_DIR ${CMAKE_INSTALL_PREFIX})
+  set(SETUP_DIR ${CMAKE_INSTALL_PREFIX}/etc/ros)
 
   if(CATKIN_INSTALL_INTO_PREFIX_ROOT)
     # install empty workspace marker if it doesn't already exist
@@ -90,7 +90,7 @@
       @ONLY)
     catkin_install_python(PROGRAMS
       ${CMAKE_BINARY_DIR}/catkin_generated/installspace/_setup_util.py
-      DESTINATION ${CMAKE_INSTALL_PREFIX})
+      DESTINATION ${SETUP_DIR})
   endif()
 
   if(NOT WIN32)
@@ -102,7 +102,7 @@
     if(CATKIN_INSTALL_INTO_PREFIX_ROOT)
       install(PROGRAMS
         ${CMAKE_BINARY_DIR}/catkin_generated/installspace/env.sh
-        DESTINATION ${CMAKE_INSTALL_PREFIX})
+        DESTINATION ${SETUP_DIR})
     endif()
     # generate and install setup for various shells
     foreach(shell bash sh zsh)
@@ -112,7 +112,7 @@
       if(CATKIN_INSTALL_INTO_PREFIX_ROOT)
         install(FILES
           ${CMAKE_BINARY_DIR}/catkin_generated/installspace/setup.${shell}
-          DESTINATION ${CMAKE_INSTALL_PREFIX})
+          DESTINATION ${SETUP_DIR})
       endif()
     endforeach()
 
